sklearn/metrics/_classification.py

```diff
@@ -2350,6 +2350,21 @@ def balanced_accuracy_score(y_true, y_pred, *, sample_weight=None, adjusted=Fals
     return score
 
 
+@validate_params(
+    {
+        "y_true": ["array-like", "sparse matrix"],
+        "y_pred": ["array-like", "sparse matrix"],
+        "labels": ["array-like", None],
+        "target_names": [list, None],
```

Comment:
It seems that our test support more cases than just list. So it could be a hidden feature but we can support officially array-like.

```suggestion
        "target_names": ["array-like", None],
```

Could you update the documentation accordingly below?

Reply:
Thanks for the suggested change, I juste implemented it and updated the docstring accordingly.